File: le-transform/src/main/java/com/latticeengines/transform/v2_0_25/functions/StdVisidbDsTitleChannel.java
package com.latticeengines.transform.v2_0_25.functions;

import java.util.LinkedHashMap;
import java.util.Map;

import com.latticeengines.transform.exposed.RealTimeTransform;
import com.latticeengines.transform.exposed.metadata.ApprovedUsage;
import com.latticeengines.transform.exposed.metadata.Category;
import com.latticeengines.transform.exposed.metadata.FundamentalType;
import com.latticeengines.transform.exposed.metadata.StatisticalType;
import com.latticeengines.transform.exposed.metadata.Tag;
import com.latticeengines.transform.exposed.metadata.TransformMetadata;
import com.latticeengines.transform.v2_0_25.common.DSUtils;

public class StdVisidbDsTitleChannel implements RealTimeTransform {

    private static final long serialVersionUID = -2669053366227213576L;
    @SuppressWarnings("rawtypes")
    private static LinkedHashMap mapTitleChannel = null;
    
    public StdVisidbDsTitleChannel() {
        
    }

    public StdVisidbDsTitleChannel(String modelPath) {

    }

    @SuppressWarnings({ "unchecked", "rawtypes" })
    @Override
    public Object transform(Map<String, Object> arguments, Map<String, Object> record) {
        String column = (String) arguments.get("column");
        Object o = record.get(column);

        if (o == null)
            return "0.0";

        if (mapTitleChannel == null) {
            mapTitleChannel = new LinkedHashMap();
            mapTitleChannel.put("Consumer", "consumer,retail");
            mapTitleChannel.put("Government", "government");
            mapTitleChannel.put("Corporate", "enterprise,corporate");
        }

        String s = (String) o;

        return DSUtils.valueReturn(s, mapTitleChannel);
    }

    @Override
    public TransformMetadata getMetadata() {
        TransformMetadata metadata = new TransformMetadata();
        metadata.setApprovedUsage(ApprovedUsage.MODEL_ALLINSIGHTS);
        metadata.setCategory(Category.LEAD_INFORMATION);
        metadata.setFundamentalType(FundamentalType.ALPHA);
        metadata.setStatisticalType(StatisticalType.NOMINAL);
        metadata.setDescription("Title Channel");
        metadata.setDisplayName("Title Channel");
        metadata.setTags(Tag.INTERNAL_TRANSFORM);
        
        return metadata;
    }

}

File: le-serviceflows/core/src/main/resources/com/latticeengines/serviceflows/core/transforms/std_visidb_ds_title_channel.py
#This function looks for words and returns a value if there is a word map
#The function is all based on mapTitleChannelAny

#required import
from POC10Functions import *

def metadata():
    return {  'ApprovedUsage'   : 'ModelAndAllInsights'
            , 'Category'        : 'Lead Information'
            , 'FundamentalType' : 'nominal'
            , 'StatisticalType' : 'nominal'
            , 'Description'     : 'Title Channel'
            , 'DisplayName'     : 'Title Channel'                                    
            , 'Tags'            : 'Internal' }

mapTitleChannelAny=[('Consumer',set(['consumer','retail'])),
                    ('Government',set(['government'])),
                    ('Corporate',set(['enterprise','corporate']))]

 
def std_visidb_ds_title_channel(x):
    return valueReturn(x,mapTitleChannelAny)

#function to call
def transform(args, record):
    column = args["column"]
    value = record[column]
    return std_visidb_ds_title_channel(value)
